tests/simulator/test_advance.py

```diff
@@ -0,0 +1,240 @@
+                FluidDiagnostics(quantity=quantity,
+                                 write_timestamps=np.zeros(advances+1),
+                                 compute_timestamps=np.zeros(advances+1),
+                                 population_name=pop)
+
+        simulator = Simulator(global_vars.sim).initialize()
+        for i in range(advances):
+            simulator.advance()
+
+        eb_hier = None
+        if qty in ["e", "eb"]:
+            eb_hier = hierarchy_from(h5_filename=diag_outputs+"/EM_E.h5", hier=eb_hier)
+        if qty in ["b", "eb"]:
+            eb_hier = hierarchy_from(h5_filename=diag_outputs+"/EM_B.h5", hier=eb_hier)
+        if qty in ["e", "b", "eb"]:
+            return eb_hier
+
+        if qty == "moments":
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_density.h5")
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_bulkVelocity.h5", hier=mom_hier)
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_density.h5", hier=mom_hier)
+            mom_hier = hierarchy_from(h5_filename=diag_outputs+"/ions_pop_protons_flux.h5", hier=mom_hier)
+            return mom_hier
+
+
+
+
+    def _test_field_coarsening_via_subcycles(self, dim, interp_order, refinement_boxes):
+        print("test_field_coarsening_via_subcycles for dim/interp : {}/{}".format(dim, interp_order))
+
+        from tests.amr.data.field.coarsening.test_coarsen_field import coarsen
+
+        advances = 3
+        stepDiff = .1
+        nSubcycles = 10
+        coarsestTimeStep = 0.001
+        levelNumbers = [i for i in range(len(refinement_boxes.keys()) + 1)]
+
+        lvlSteps = [coarsestTimeStep * (stepDiff ** (ilvl)) for ilvl in levelNumbers]
+        finestTimeStep = lvlSteps[-1]
+        secondFinestTimeStep = lvlSteps[-2] # this test makes no sense with only 1 level
+        totalSteps = nSubcycles ** levelNumbers[-1] * advances
+        uniqTimes = set([0])
+
+        diag_outputs=f"phare_outputs_subcycle_coarsening_{self.ddt_test_id()}"
+        datahier = self.getHierarchy(interp_order, refinement_boxes, "eb", cells=30,
+                                      diag_outputs=diag_outputs,
+                                      extra_diag_options={"fine_dump_lvl_max": 10},
+                                      advances=advances, smallest_patch_size=5, largest_patch_size=30)
+
+        times = datahier.times()
+        for step in range(1, totalSteps + 1):
+            checkTime = float("{:.6f}".format(finestTimeStep * step))
+            self.assertIn(checkTime, times)
+            uniqTimes.add(checkTime)
+
+        self.assertEqual(len(uniqTimes), len(datahier.time_hier.items()))
+
+        syncSteps = nSubcycles ** levelNumbers[-2] * advances # ignore most fine subcycles
+
+        # FIX THIS AFTER NO MORE REGRIDS
+        #  SEE: https://github.com/PHAREHUB/PHARE/issues/400
+        assert syncSteps % advances == 0 # perfect division
```

Comment:
do we need to make sure syncSteps and advances are integers before doing the % ?

Reply:
I don't think so, % works on floats